### PR TITLE
General fixes and changes to sync with the work done in Online Services

### PR DESCRIPTION
**Pull request by caiobegotti**
I understand some of these changes might be just cosmetics and not entirely desirable but here they all are:

- Jenkins root-disk minimal size had to be bumped so I could test multiple builds in the same deployment
- Uses env-template for variable substitution, although I think both replace-image-tag.sh and deploy-to-kubernetes.sh could end up being a single "control" script later.
- The DOMAIN var will be sorted later but it is passed on like the others
- IMAGE_TAG is actually the Git commit hash now, to better track what had been deployed

See https://pastebin.canonical.com/181433/ for a "workflow" after it is all done:

1. Commit/push to the app repo
2. Tested the dev env for the specific Git commit hash
3. Trigger the Jenkins job passing "qa" as namespace (it took 7 seconds for it to be up and live)
4. QA'ed changes and did the same as before but passing "prod" as namespace
5. Celebrated

### CODE REVIEW

**Comment by caiobegotti** [comment]
Please squash it before doing anything with those changes. My apologies for the dirty commit history this time, ugh.

File: setup.sh [hunk 1]
```diff
@@ -65,12 +64,13 @@ juju switch kubernetes-${KUBE_RANDOM}
 juju deploy canonical-kubernetes
 
 juju switch jenkins-${JENK_RANDOM}
-juju deploy bundle.yaml
+juju deploy jenkins.yaml
```
Comment (lazypower) [inline setup.sh:67]: +1 on the topical rename here.

File: workspaces/deploy-to-kubernetes/config.xml [hunk 2]
```diff
@@ -7,9 +7,19 @@
     <hudson.model.ParametersDefinitionProperty>
       <parameterDefinitions>
         <hudson.model.StringParameterDefinition>
+          <name>IMAGE_OWNER</name>
```
Comment (lazypower) [inline workspaces/deploy-to-kubernetes/config.xml:10]: Ah interesting! parameterized the build with registry details. +1

File: workspaces/mm-container-build/config.xml [hunk 3]
```diff
@@ -62,7 +62,7 @@ mv mattermost/* app/app_code/</command>
       <dockerfilePath>app/Dockerfile</dockerfilePath>
       <skipBuild>false</skipBuild>
       <skipDecorate>false</skipDecorate>
-      <repoTag>${BUILD_NUMBER}</repoTag>
+      <repoTag>${GIT_COMMIT}</repoTag>
```
Comment (lazypower) [inline workspaces/mm-container-build/config.xml:65]: +1 on this identifier moving through. I like the idea of having the git commit be the unique identifier, until we get into having multiple revisions of that single git commit in all our ci infra, and trying to decouple what happened where. I think there's room for improvement here but I don't have a pliable solution today.

Perhaps this works fine as is and I'm over-thinking it too :)

**Review by lazypower — CHANGES_REQUESTED** [review]
I see some good work in here. Once you've squashed the commit history I'll be happy to approve and merge.

**Comment by caiobegotti** [comment]
Done, just please don't forget about https://bitbucket.org/jujudemo/mattermost-deploy/pull-requests/1/sync-work-done-for-a-similar-app-in-online/diff too, as they are hand-in-hand :bowtie:

**Comment by lazypower** [comment]
+1 LGTM

**Comment by mbruzek** [comment]
I also looked over these changes and was unable to get my +1 before Chuck merged it!

**Pull request merged**